### PR TITLE
Use markdown for URLs in Discord publications

### LINKED ISSUE #1681: discord notifications: put URL inside the message text
Current notification for wiki edit:
> Page InternalSystem/PublicationContent/M5436 edited by adelikat (use relative link) https://tasvideos.org/InternalSystem/PublicationContent/M5436

Discord allows markdown links now, so we can do this instead:
> Page [InternalSystem/PublicationContent/M5436](https://tasvideos.org/InternalSystem/PublicationContent/M5436) edited by adelikat (use relative link)

For forum:
> [New Post](https://tasvideos.org/Forum/Posts/525070) by feos (WorkB: #8578: Mizumaririn's NES Super Mario Bros. 3Mix "all 300 star coins" in 2:07:17.64)

Catalog:
> [8519S](https://tasvideos.org/8519S) Catalog edited by feos (Game changed from "" to Sonic 1: Return to the Origin, Game Version changed from "" to Sonic 1: Return to the Origin | #8519: CyanTheGamerhog01's Genesis Sonic 1: Return to the Origin "glitchless" in 19:23.82)

etc

This would make the log less cluttered and reduce copypaste.

### PR DESCRIPTION
**Pull request by qixils**
Future work could expand on this with more URLs (say, linking to the responsible users) but figured this might be a good start to reduce visual clutter in the Discord update channels. Not sure if this might be considered too platform-specific to embed as a variable in Post? I figure markdown is relatively platform-agnostic so it's not too bad but 🤷🏻‍♀️ 

edit: closes #1681

### CODE REVIEW

File: TASVideos/Pages/Forum/Posts/Edit.cshtml.cs [hunk 1]
```diff
@@ -233,6 +234,7 @@ public async Task<IActionResult> OnPostDelete()
 			await _publisher.SendForum(
 				post.Topic!.Forum!.Restricted,
 				$"{(topicDeleted ? "Topic" : "Post")} DELETED by {User.Name()}",
+				$"[{(topicDeleted ? "Topic" : "Post")} DELETED]({{0}}) by {User.Name()}",
 				$"{post.Topic!.Forum!.ShortName}: {post.Topic.Title}",
```
Comment (qixils) [inline TASVideos/Pages/Forum/Posts/Edit.cshtml.cs:238]: I think ideally this link would go on `{post.Topic.Title}` but I was hesitant to create *another* variable just for this one use, makes me wonder if there's some better way to design this-- possibly creating a new class to store plaintext and markdown (and maybe HTML/others?) variants of these strings? But then I'm not sure if that'd be too invasive of a change so I dunno, feedback is welcome on this front.

File: TASVideos/Pages/Games/Edit.cshtml.cs [hunk 2]
```diff
@@ -195,7 +203,11 @@ public async Task<IActionResult> OnPostDelete()
 		var saveResult = await ConcurrentSave(_db, saveMessage, $"Unable to delete Game {Id}");
 		if (saveResult)
 		{
-			await _publisher.SendGameManagement($"{saveMessage} by {User.Name()}", "", $"{Id}G");
+			await _publisher.SendGameManagement(
+				$"{saveMessage} by {User.Name()}",
+				"",
+				"",
+				"");
```
Comment (qixils) [inline TASVideos/Pages/Games/Edit.cshtml.cs:210]: Removed 404 link

File: TASVideos/Pages/Roles/AddEdit.cshtml.cs [hunk 3]
```diff
@@ -145,7 +145,8 @@ public async Task<IActionResult> OnPostDelete()
 			await _publisher.SendUserManagement(
 				$"Role {Id} deleted by {User.Name()}",
 				"",
-				"Roles/List");
+				"",
+				"");
```
Comment (qixils) [inline TASVideos/Pages/Roles/AddEdit.cshtml.cs:149]: Removed sorta irrelevant link

**Comment by Randomno** [comment]
Would this close #1681?

**Comment by qixils** [comment]
Looks like it should, I'll note it in the body, thanks.

File: TASVideos.Core/Services/ExternalMediaPublisher/ExternalMediaPublisher.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ public string ToAbsolute(string relativeLink)
 
 public static class ExternalMediaPublisherExtensions
 {
-	public static async Task SendUserFile(this ExternalMediaPublisher publisher, bool unlisted, string title, string relativeLink, string body)
+	public static async Task SendUserFile(this ExternalMediaPublisher publisher, bool unlisted, string title, string formattedTitle, string relativeLink, string body)
```
Comment (adelikat) [inline TASVideos.Core/Services/ExternalMediaPublisher/ExternalMediaPublisher.cs:44]: would markdownTitle be a better name for this?

**Review by adelikat — APPROVED** [review]
Going to merge, and rename formattedTitle in a future commit

**Pull request merged**